src/modules/create_institution/app/create_institution_usecase.ts

```diff
@@ -0,0 +1,64 @@
+import { randomUUID } from 'crypto';
+import { Institution } from "../../../core/structure/entities/Institution";
+import { MissingParameter, UserNotAuthenticated } from '../../../core/helpers/errors/ModuleError';
+import { IInstitutionRepo } from '../../../core/repositories/interfaces/IInstitutionRepo';
+import { IUserRepo } from '../../../core/repositories/interfaces/IUserRepo';
+import { UserTypeEnum } from '../../../core/helpers/enums/UserTypeEnum';
+import jwt from 'jsonwebtoken';
+
+export class CreateInstitutionUsecase {
+    public database_repo: IInstitutionRepo;
+    public user_repo: IUserRepo;
+
+    constructor(database_repo: IInstitutionRepo, user_repo: IUserRepo) {
+        this.database_repo = database_repo;
+        this.user_repo = user_repo;
+    }
+
+    public async execute(institutionData: any, headers: any){
+        if (!headers.authorization) {
+            throw new MissingParameter("Authorization");
+        }
+        
+        const decodedToken = jwt.verify(headers.authorization, process.env.JWT_SECRET as string);
+        const userId = decodedToken;
+        
+        
+        const userExists = await this.user_repo.get_user(headers.authorization);
+        if (!userExists || userExists.user_type !== UserTypeEnum.ADMIN && userExists.user_type !== UserTypeEnum.MODERATOR) {
+            throw new UserNotAuthenticated("User is not authenticated");
+        }
```

Comment:
Check how u do the logic in this usecase: [repo](https://github.com/Instituto-Maua-de-Tecnologia/coil_mss/blob/creating-get-activity/src/modules/get_activity/app/get_activity_usecase.ts)

Lines 40 to 51